storage/aptosdb/src/state_store/mod.rs

```diff
@@ -284,6 +284,44 @@ impl StateStore {
         max_nodes_per_lru_cache_shard: usize,
         hack_for_tests: bool,
     ) -> Self {
+        Self::sync_commit_progress(
+            Arc::clone(&ledger_db),
+            Arc::clone(&state_kv_db),
+            /*crash_if_difference_is_too_large=*/ true,
+        );
+        let state_merkle_db = Arc::new(StateMerkleDb::new(
+            state_merkle_db,
+            max_nodes_per_lru_cache_shard,
+        ));
+        let state_db = Arc::new(StateDb {
+            ledger_db,
+            state_merkle_db,
+            state_kv_db,
+            state_merkle_pruner,
+            epoch_snapshot_pruner,
+            state_kv_pruner,
+        });
+        let buffered_state = Mutex::new(
+            Self::create_buffered_state_from_latest_snapshot(
+                &state_db,
+                buffered_state_target_items,
+                hack_for_tests,
+                /*check_max_versions_after_snapshot=*/ true,
+            )
+            .expect("buffered state creation failed."),
+        );
+        Self {
+            state_db,
+            buffered_state,
+            buffered_state_target_items,
+        }
+    }
+
+    pub fn sync_commit_progress(
```

Comment:
nit: would be helpful to add some comments on why truncating ledger and state_kv db to overall progress to keep them in sync.

Reply:
Done.